substrate/primitives/metadata-ir/src/types.rs

```diff
@@ -361,19 +374,29 @@ impl From<MetaType> for PalletCallMetadataIR {
 pub struct PalletEventMetadataIR<T: Form = MetaForm> {
 	/// The Event type.
 	pub ty: T::Type,
+	/// Deprecation status of the event itself
+	pub deprecation_info: DeprecationStatus<T>,
+	/// Deprecation status of the variants
+	pub deprecated_variants: BTreeMap<T::String, DeprecationStatus<T>>,
```

Comment:
nit: Would the parsing be more complex if we propagate here a variant index (`usize`) instead of variant name? Just to keep consistency with the calls

Reply:
hmm, i think it should be fine.

Reply:
pushed the change with usize indexes instead of strings